test/v2/controllers/requestParsers/validators/SubmitForeignPropertyBsasValidatorSpec.scala

```diff
@@ -0,0 +1,823 @@
+            |            "other": 123.45
+            |        }
+            |    },
+            |    "foreignFhlEea": {
+            |        "income": {
+            |            "rentIncome": 123.456
+            |        },
+            |        "expenses": {
+            |            "premisesRunningCosts": 123.456,
+            |            "repairsAndMaintenance": 123.45,
+            |            "financialCosts": 123.45,
+            |            "professionalFees": 123.45,
+            |            "costOfServices": 123.45,
+            |            "travelCosts": 123.45,
+            |            "other": 123.45
+            |        }
+            |    }
+            |}
+            |""".stripMargin)
+        validator.validate(SubmitForeignPropertyRawData(validNino, validBsasId, badjson)) shouldBe List(
+          formatError("/foreignProperty/income/rentIncome"),
+          formatError("/foreignProperty/expenses/premisesRunningCosts"),
+          formatError("/foreignFhlEea/income/rentIncome"),
+          formatError("/foreignFhlEea/expenses/premisesRunningCosts"))
+      }
+    }
+    "return a RULE_RANGE_INVALID error" when {
+      "a value field is above 99999999999.99" in {
+        val badJson = Json.parse(
+          """
+            |{
+            |    "foreignProperty": {
+            |        "income": {
+            |            "rentIncome": 100000000000.99,
+            |            "premiumsOfLeaseGrant": 123.12,
+            |            "foreignTaxTakenOff": 123.12,
+            |            "otherPropertyIncome": 123.12
+            |        },
+            |        "expenses": {
+            |            "premisesRunningCosts": 123.12,
+            |            "repairsAndMaintenance": 123.12,
+            |            "financialCosts": 123.12,
+            |            "professionalFees": 123.12,
+            |            "travelCosts": 123.12,
+            |            "costOfServices": 123.12,
+            |            "residentialFinancialCost": 123.12,
+            |            "other": 123.12
+            |        }
+            |    },
+            |    "foreignFhlEea": {
+            |        "income": {
+            |            "rentIncome": 123.12
+            |        },
+            |        "expenses": {
+            |            "premisesRunningCosts": 123.12,
+            |            "repairsAndMaintenance": 123.12,
+            |            "financialCosts": 123.12,
+            |            "professionalFees": 123.12,
+            |            "costOfServices": 123.12,
+            |            "travelCosts": 123.12,
+            |            "other": 123.12
+            |        }
+            |    }
+            |}
+            |""".stripMargin)
+        validator.validate(SubmitForeignPropertyRawData(validNino, validBsasId, badJson)) shouldBe List(rangeError("/foreignProperty/income/rentIncome"))
+      }
+      "multiple fields are below 0" in {
```

Comment:
No fields here are below 0?

Reply:
Also the minimum value is -99999999999.99 not 0 - for something to be below the minimum it'd have to be beyond -1000000000000.00